### DISCUSSION TITLE
Can MySQL strict mode be disabled at session level ?

### DISCUSSION THREAD

**Discussion by OsObO856** [OS4ED/openSIS-Classic#300] [Q&A]
Dear all. 

I'm trying to install OpenSIS on a containerized LAMP running in [Cloudron](https://docs.cloudron.io/apps/lamp/). 

OpenSIS fails to install because the global default sql_mode in Mysql is set to **strict mode**

![StrictMode](https://github.com/OS4ED/openSIS-Classic/assets/126863894/baca0bb5-ba05-4145-8ff5-1bd50adad8ef)

Due to the specific nature of Cloudron it is not possible by design to change this setting . Any change in the MySQL configuration file my.cnf would break the container.

I understand from reading a [forum discussion on this matter](https://forum.cloudron.io/topic/2631/disable-strict-mode-mysql/2) that sql_mode can, however, be set per session with a workaround. In the PHP code, where you create the database connection it is advised to query  something like     **SET sql_mode = ' ' ;**

Unfortunately I'm not a dev and have no clue to where to add this instruction. Could you please suggest me a way ? 
Thanks in advance for the support !
Best wishes 
Bertrand

**Comment by kochinc** [comment]
I just installed openSIS on my own server for evaluation, and got the exact message while going through the installation process. In order to get through the installation, I added "SET sql_mode=''" to the first few PHP scripts in install/ folder. With the modifications I was able to continue with installation. However, soon I realized there are copy-and-paste codes in many scripts for database connections, and decided not to go on with the evaluation.

To make it work, one has 3 choices: cleaning up the MySQL codes, refactoring the database connection codes, or adding "SET sql_mode=''" everywhere. I chose to move on to evaluate other SIS projects.
